remix/app/routes/newsletters.tsx

```diff
@@ -0,0 +1,61 @@
+import type { User } from "@prisma/client";
+import type { LoaderArgs } from "@remix-run/node";
+import {
+  Form,
+  isRouteErrorResponse,
+  useLoaderData,
+  useRouteError,
+} from "@remix-run/react";
+import { SocialsProvider } from "remix-auth-socials";
+import { authenticator } from "~/services/auth.server";
+import type { UserProfile } from "~/types";
+import { db } from "~/utils/db.server";
+
+export const loader = async ({ request }: LoaderArgs) => {
+  const userRequest = (await authenticator.isAuthenticated(request)) as any;
+  const userProfile: UserProfile = userRequest?._json;
+  if (!userProfile) {
+    throw new Response("Forbidden", { status: 403 });
+  }
+  let user: User | null = null;
+
+  try {
+    user = await db.user.findUnique({
+      where: { email: userProfile.email },
+    });
+  } catch (err) {
+    console.log(err);
```

Comment:
Let's avoid using `console.log` (as it is unsafe and we tend to forget removing it) and instead use `throw` so that it can be caught by the `ErrorBoundary`

Reply:
Also shall we use https://remix.run/docs/en/main/route/error-boundary-v2?

Reply:
Yes the error boundary I have already follow the V2 version 😄.